lib/sablon/html/ast.rb

```diff
@@ -10,6 +10,54 @@ def self.node_name
       end
     end
 
+    class NodeProperties
+      def inspect
+        @properties.map { |k, v| v ? "#{k}=#{v}" : k }.join(';')
+      end
+
+      def [](key)
+        @properties[key]
+      end
+
+      def []=(key, value)
+        @properties[key] = value
+      end
+
+      def to_docx(parent_tag)
```

Comment:
I don't like the fact that this method has a different signature than other `to_docx` methods. All of this code passes `'w:p'`. Why not go with `ParagraphProperties` and if necessary extend from a more abstract `NodeProperties` class (only necessary once we see more specializations). Another approach would be to keep `NodeProperties` but pass this key when the object is instantiated. To simplify things for call-sites we can provide factory methods like `.paragraph`.

```ruby
NodeProperties.paragraph({})
```